src/agent/border_agent.cpp

```diff
@@ -63,7 +63,8 @@
 
 namespace otbr {
 
-static const char kBorderAgentServiceType[] = "_meshcop._udp"; ///< Border agent service type of mDNS
+static const char kBorderAgentServiceType[] = "_meshcop._udp";           ///< Border agent service type of mDNS
+static const char kBorderAgentServiceName[] = OTBR_MESHCOP_SERVICE_NAME; ///< Border agent service name of mDNS
```

Comment:
```suggestion
static const char kBorderAgentServiceInstanceName[] = OTBR_MESHCOP_SERVICE_INSTANCE_NAME; ///< Border agent service name of mDNS
```
Prefer to add `Instance` for clarity. 
I think `service name` is quite misleading. RFC 6763 refers to service types (e.g. `_meshcop._udp`) as the service names, and always use `instance name` for service instance names.